### ISSUE TITLE
[rush] rushx does not escape CLI parameters properly

### ISSUE THREAD

**Issue by octogonz** [microsoft/rushstack#2695]
## Repro steps

Create a **package.json** file like this:

**package.json**
```js
{
  "scripts": {
    "action": "node action.js"
  }
}
```
...which invokes:

**action.js**
```js
console.log(JSON.stringify(process.argv, undefined, "  "));
```

Then invoke `rushx action a b c\&d`

The output is:
```
Rush Multi-Project Build Tool 5.46.1 (unmanaged) - https://rushjs.io
Node.js version is 12.17.0 (LTS)

Executing: "node test.js a b c&d"

[
  "C:\\Program Files\\nodejs\\node.exe",
  "C:\\vb\\test.js",
  "a",
  "b",
  "c"
]
'd' is not recognized as an internal or external command,
operable program or batch file.
```

  **Expected result:**  The `c&d` argument should be escaped rather than expanded by the shell.

Note that `npm run action -- a b c\&d` does the right thing.

## Details

The problem is here:
https://github.com/microsoft/rushstack/blob/45183334665dc7fc9495ed14900b7c7f519b53e9/apps/rush-lib/src/cli/RushXCommandLine.ts#L106-L109

## Standard questions

Please answer these questions to help us investigate your issue more quickly:

| Question | Answer |
| -------- | -------- |
| `@microsoft/rush` globally installed version? | 5.46.1 |
| `rushVersion` from rush.json? | 5.46.1 |
| `useWorkspaces` from rush.json? | true |
| Operating system? | Windows + Mac  |
| Would you consider contributing a PR? |Yes |

**Comment by octogonz** [comment]
Looks like `pnpm run` simply adds `"` around the arguments:

[/packages/lifecycle/src/runLifecycleHook.ts](https://github.com/pnpm/pnpm/blob/a468d2b3f8e4a456b3e57ad8a013af65b29c0484/packages/lifecycle/src/runLifecycleHook.ts#L38)
```ts
  if (opts.args?.length && m.scripts?.[stage]) {
    m.scripts[stage] = `${m.scripts[stage]} ${opts.args.map((arg) => `"${arg}"`).join(' ')}`
  }
```

**Comment by octogonz** [comment]
After some more testing, I found that `pnpm run` mishandles this case:
```bash
$ pnpm run action a b\" c
```
whereas `npm run` handles it correctly.  NPM does this escaping:

[@npmcli/run-script/lib/run-script-pkg.js](https://github.com/npm/run-script/blob/47a4d539fb07220e7215cc0e482683b76407ef9b/lib/run-script-pkg.js#L43)
```js
  else if (event === 'start' && await isServerPackage(path))
    cmd = 'node server.js' + args.map(a => ` ${JSON.stringify(a)}`).join('')
```

This is a better solution.

**Comment by octogonz** [comment]
Released with Rush 5.47.0
